tests/pytorch/r1.13/dlrm.libsonnet

```diff
@@ -82,94 +83,126 @@ local utils = import 'templates/utils.libsonnet';
-    --memory-map \
-    --arch-sparse-feature-size=16 \
-    --arch-mlp-bot="13-512-256-64-16" \
-    --arch-mlp-top="512-256-1" \
-    --data-generation=dataset \
-    --data-set=kaggle \
-    --loss-function=bce \
-    --round-targets=True \
-    --learning-rate=0.1 \
-    --mini-batch-size=128 \
-    --print-freq=1024 \
-    --print-time \
-    --test-mini-batch-size=16384 \
-    --test-freq=101376 \
-    --use-tpu \
-    --num-indices-per-lookup=1 \
-    --num-indices-per-lookup-fixed \
-    --tpu-model-parallel-group-len 8 \
-    --tpu-metrics-debug \
-    --tpu-cores=8 |& tee dlrm_logs.txt
-    acc=`grep Testing dlrm_logs.txt | tail -1 | grep -oP 'best \K[+-]?([0-9]*[.])?[0-9]+'`
-    echo 'Accuracy is' $acc
-    test $(echo $acc'>'78.75 | bc -l) -eq 1  # assert cls acc higher than 78.75
-  |||,
+
   local criteo_kaggle = common.Convergence {
+    paramsOverride+:: {
+      miniBatchSize: 128,
+      archSparseFeatureSize: 16,
+      archMlpBot: '13-512-256-64-16',
+      archAlpTop: '512-256-1',
+      numIndicesPerLookup: 1,
+      dataGeneration: 'dataset',
+      trainCommand+: [
+        '--raw-data-file=/datasets/criteo-kaggle-mm/train.txt',
+        '--processed-data-file=/datasets/criteo-kaggle-mm/kaggleAdDisplayChallenge_processed.npz',
+        '--memory-map',
+        '--print-time',
+        '--test-mini-batch-size=16384',
+        '--test-freq=101376',
+        '--data-set=kaggle',
+        '--loss-function=bce',
+        '--round-targets=True',
+        '--learning-rate=0.1',
+        '--print-freq=1024',
+        '--no-save',
+        '--nepochs=25',
+      ],
+    },
     command: utils.scriptCommand(
       |||
+        set +e
         apt-get install -y bc
         pip install onnx
         git clone --recursive https://github.com/pytorch-tpu/examples.git
-        python examples/deps/dlrm/dlrm_tpu_runner.py \
-          %(convergence_common)s
-      ||| % convergence_common
+        %s
+      ||| % utils.toCommandString(self.paramsOverride.trainCommand),
     ),
   },
+
+  local tpuVm = common.PyTorchTpuVmMixin {
+    tpuSettings+: {
+      tpuVmExtraSetup: |||
+        pip3 install tqdm sklearn tensorboardX google-cloud-storage
+        git clone -b tpu-xrt --single-branch https://github.com/darisoy/dlrm.git dlrm-xrt/
+        echo 'export PATH=~/.local/bin:$PATH' >> ~/.bash_profile
+      |||,
+    },
+    paramsOverride+: {
+      scriptPath: 'dlrm-xrt/dlrm_tpu_runner.py',
+    },
+  },
+
+  local pjrt = tpuVm + experimental.PjRt {
+    modelName: 'dlrm-pjrt',
+  },
+
   local v3_8 = {
     accelerator: tpus.v3_8,
   },
+  local v4_8 = {
+    accelerator: tpus.v4_8,
+  },
   configs: [
     dlrm + v3_8 + one_core + timeouts.Hours(3) + mixins.Experimental,
     dlrm + v3_8 + seq_fwd + timeouts.Hours(3) + mixins.Experimental,
     dlrm + v3_8 + mp_fwd + timeouts.Hours(3) + mixins.Experimental,
     dlrm + v3_8 + mp_dp_fwd + timeouts.Hours(3),
-    dlrm_convergence + v3_8 + criteo_kaggle + timeouts.Hours(6),
```

Comment:
did we remove convergence here in purpose?

Reply:
hm, this convergence doesn't appear in the tests/pytorch/nightly/dlrm.libsonnet: https://github.com/GoogleCloudPlatform/ml-testing-accelerators/blob/df347546f2ff0c71ac7acbc25babdf9ef985216d/tests/pytorch/nightly/dlrm.libsonnet#L200-L206. Do you want me to add it?

Reply:
ok I guess we remove it from nightly for some reason, I think it is OK.